src/bin/cli.rs

```diff
@@ -186,18 +197,18 @@ fn binary_prompt() -> Result<BinaryOperation, InquireError> {
 /// # Panics
 ///
 /// Panics if `n` is greater than `size`.
-fn qubit_prompt(n: usize, size: usize) -> Result<Vec<usize>, InquireError> {
+fn qubit_prompt<const N: usize>(target_names: [&str; N], size: usize) -> Result<Vec<usize>, InquireError> {
```

Comment:
I suggest we handle this in a different way than const generics, as the feature was not designed with this functionality in mind